extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergReadInstructions.java

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.annotations.BuildableStyle;
+import org.immutables.value.Value.Immutable;
+
+import java.util.Map;
+
+/**
+ * This class provides instructions intended for reading Iceberg catalogs and tables. The default values documented in
+ * this class may change in the future. As such, callers may wish to explicitly set the values.
+ */
+@Immutable
+@BuildableStyle
+public abstract class IcebergReadInstructions implements IcebergBaseInstructions {
+    /**
+     * The default {@link IcebergReadInstructions} to use when reading Iceberg data files. Providing this will use
+     * system defaults for cloud provider-specific parameters.
+     */
+    public static final IcebergReadInstructions DEFAULT = builder().build();
+
+    public static Builder builder() {
+        return ImmutableIcebergReadInstructions.builder();
+    }
+
+    /**
+     * A {@link Map map} of rename instructions from Iceberg to Deephaven column names to use when reading the Iceberg
+     * data files.
+     */
+    public abstract Map<String, String> columnRenames();
```

Comment:
I know this was just pulled up from Base instructions, but whenever we have an Iceberg column name, it should ideally be tied to a Schema, I mention as much in https://github.com/deephaven/deephaven-core/issues/6124. I don't know if this is actionable here, and it may be more of a discussion for https://github.com/deephaven/deephaven-core/pull/5707

Reply:
Same response as https://github.com/deephaven/deephaven-core/pull/5989#discussion_r1792187149